### DISCUSSION TITLE
Editable commits author and commiter emails

### DISCUSSION THREAD

**Discussion by tiberium** [jesseduffield/lazygit#3977] [Ideas]
Would be nice to select in the CommitMenu the author's email. Even better if it would be possible to set the committers email as well. In git this is of course taken from the git config, but can be overwritten by setting the `GIT_AUTHOR_EMAIL` env (same for committer: `GIT_COMMITTER_EMAIL`).

Many of us have many emails attached to ours git users, and selecting the right one for a specific commit would be great.

**Comment by mark2185** [comment]
You can press `a` on a commit in the `Commits` panel, does that help?
![image](https://github.com/user-attachments/assets/b4fb591c-2b21-4c6b-b6c7-3b4b5fc6e619)

**Reply by tiberium** [reply]
This works on existing commits, right? I am rather thinking about setting the author/committer while committing a new commit.

**Comment by stefanhaller** [comment]
@tiberium I'd be interested in the use case for this. Do you have workflows where you commit with different email addresses _within the same repo_? I have never seen this, I use my work email for some repos and my private email for others.

You _are_ aware that you can set the git config per repo, aren't you?

**Reply by tiberium** [reply]
Yes, I know that I can set git config per repo. The use case is: I work with many repositories, and not all of them are configured locally, some of them were cloned by me only to do a small change, and few times I have already by accident used a wrong email. Such a feature would simply speed-up committing to a repo without creating the config for it.

**Reply by stefanhaller** [reply]
Honestly, this seems like so much of a niche use case to me that I'd be against complicating lazygit's user interface to accommodate it. I feel that fixing the author after the fact like Luka suggested above is a good enough solution to this.
